src/swapd/swap_state.rs

```diff
@@ -75,14 +78,18 @@ pub enum BobState {
     // #[display("CoreArb: {0:#?}")]
     #[display("CoreArb")]
     CorearbB {
+        checkpoint: SwapCheckpointType,
         received_refund_procedure_signatures: bool,
         cancel_seen: bool,
         remote_params: Params,
         local_params: Params,
         b_address: bitcoin::Address,
     }, // lock (not signed), cancel_seen, remote
     #[display("BuySig")]
-    BuySigB { buy_tx_seen: bool },
+    BuySigB {
+        buy_tx_seen: bool,
+        checkpoint: SwapCheckpointType,
```

Comment:
`checkpoint_type`?

Reply:
when i read, I assumed the entire checkpoint was in the state

Reply:
`last_checkpoint_type` :)

Reply:
Done in f44aa830eabd6de683354f1cb197b45287943fe3